src/app/pages/BlocksPage/index.tsx

```diff
@@ -52,27 +72,52 @@ export const BlocksPage: FC = () => {
             },
           }
         },
+        keepPreviousData: tableView === TableLayout.Vertical,
       },
     },
   )
 
   return (
-    <PageLayout>
+    <PageLayout
+      mobileFooterAction={
+        tableView === TableLayout.Vertical && (
+          <LoadMoreButton pagination={pagination} isLoading={blocksQuery.isLoading} />
+        )
+      }
+    >
       {!isMobile && <Divider variant="layout" />}
-      <SubPageCard title={t('blocks.latest')}>
-        <Blocks
-          isLoading={blocksQuery.isLoading}
-          blocks={blocksQuery.data?.data.blocks}
-          limit={PAGE_SIZE}
-          verbose={true}
-          pagination={{
-            selectedPage: pagination.selectedPage,
-            linkToPage: pagination.linkToPage,
-            totalCount: blocksQuery.data?.data.total_count,
-            isTotalCountClipped: blocksQuery.data?.data.is_total_count_clipped,
-            rowsPerPage: NUMBER_OF_ITEMS_ON_SEPARATE_PAGE,
-          }}
-        />
+      <SubPageCard
+        title={t('blocks.latest')}
+        action={isMobile && <TableLayoutButton tableView={tableView} setTableView={setTableView} />}
+        noPadding={tableView === TableLayout.Vertical}
+      >
+        {tableView === TableLayout.Horizontal && (
+          <Blocks
+            isLoading={blocksQuery.isLoading}
+            blocks={blocksQuery.data?.data.blocks}
+            limit={PAGE_SIZE}
+            verbose={true}
+            pagination={{
+              selectedPage: pagination.selectedPage,
+              linkToPage: pagination.linkToPage,
+              totalCount: blocksQuery.data?.data.total_count,
+              isTotalCountClipped: blocksQuery.data?.data.is_total_count_clipped,
+              rowsPerPage: NUMBER_OF_ITEMS_ON_SEPARATE_PAGE,
+            }}
+          />
+        )}
+        {tableView === TableLayout.Vertical && (
+          <BlockDetails>
+            {blocksQuery.isLoading &&
+              [...Array(PAGE_SIZE).keys()].map(key => (
+                <BlockDetailView key={key} isLoading={true} block={undefined} standalone />
+              ))}
+            {!blocksQuery.isLoading &&
+              blocksQuery.data?.data.blocks.map(block => (
+                <BlockDetailView key={block.hash} block={block} standalone />
+              ))}
+          </BlockDetails>
+        )}
       </SubPageCard>
```

Comment:
In transactions you moved this outside `</SubPageCard>`

Reply:
(so there's a different margin after title)

|  |  |
| --- | --- |
|  ![ml-mobile-vertical-blocks oasis-explorer pages dev_emerald_blocks (1)](https://user-images.githubusercontent.com/3758846/230373023-ec076391-6538-4e88-b31f-a0a967c9b818.png) |  ![ml-mobile-vertical-blocks oasis-explorer pages dev_emerald_transactions](https://user-images.githubusercontent.com/3758846/230373030-d8c4841e-77dd-45df-8c79-2cff89b028ea.png) |